src/main/java/io/reactivex/rxjava3/core/Flowable.java

```diff
@@ -18620,4 +18619,121 @@ public final Completable switchMapCompletableDelayError(@NonNull Function<? supe
         return ts;
     }
 
+    // -------------------------------------------------------------------------
+    // JDK 8 Support
+    // -------------------------------------------------------------------------
+
+    /**
+     * Converts the existing value of the provided optional into a {@link #just(Object)}
+     * or an empty optional into an {@link #empty()} {@code Flowable} instance.
+     * <p>
+     * <img width="640" height="335" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/fromOptional.f.png" alt="">
+     * <p>
+     * Note that the operator takes an already instantiated optional reference and does not
+     * by any means create this original optional. If the optional is to be created per
+     * consumer upon subscription, use {@link #defer(Supplier)} around {@code fromOptional}:
+     * <pre><code>
+     * Flowable.defer(() -&gt; Flowable.fromOptional(createOptional()));
+     * </code></pre>
+     * <dl>
+     *  <dt><b>Backpressure:</b></dt>
+     *  <dd>The returned {@code Flowable} supports backpressure.</dd>
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>{@code fromOptional} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * @param <T> the element type of the optional value
+     * @param optional the optional value to convert into a {@code Flowable}
+     * @return the new Flowable instance
+     * @see #just(Object)
+     * @see #empty()
+     * @since 3.0.0
+     */
+    @CheckReturnValue
+    @BackpressureSupport(BackpressureKind.FULL)
+    @SchedulerSupport(SchedulerSupport.NONE)
+    public static <T> Flowable<T> fromOptional(@NonNull Optional<T> optional) {
```

Comment:
+NonNull for return type

Reply:
Done. I suppose it should be added to all other methods too.

Reply:
That plus I think Jake meant `<T> Flowable<@NonNull T>`

Reply:
Okay, the other methods will get these in some other PRs.